### ISSUE TITLE
azcopy sync doesn't work- used azcopy cp instead with --overwrite even with it recopied every data again and again.

### ISSUE THREAD

**Issue by ozgurozkan123** [Azure/azure-storage-azcopy#151]
##### Note: The version is visible when running AzCopy without any argument
azcopy version 10.0.4-Preview
### Which platform are you using? (ex: Windows, Mac, Linux)
Linux
### What command did you run?
==azsync.sh==
while :
    do
        ./azcopy sync "https://{redactedblob}.blob.core.windows.net/parse/*?{Redacted sas url}" photos/ --recursive=true >testson30.txt 2>&1
done
==========
nohup sh azsync.sh  &

### What problem was encountered?
===error start====
Job d6957aa1-3996-5a4e-4d19-67ab1bb0a8e5 has started

Log file is located at: /root/.azcopy/d6957aa1-3996-5a4e-4d19-67ab1bb0a8e5.log
^M0 File Scanned at Source, 0 Files Scanned at Destination^M0 File Scanned at Source, 0 Files Scanned at Destination^M0 File Scanned at Source, 0 Files Scanned at Destination^M0 File Scanned at Source, 264664 Files Scanned at Destination^M0 File Scanned at Source, 578519 Files Scanned at Destination^M0 File Scanned at Source, 862428 Files Scanned at Destination^M0 File Scanned at Source, 1191110 Files Scanned at Destination^M0 File Scanned at Source, 1511732 Files Scanned at Destination^M0 File Scanned at Source, 1742640 Files Scanned at Destinationpanic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/Azure/azure-storage-azcopy/cmd.(*syncDownloadEnumerator).listSourceAndCompare(0xc420672000, 0xc420650000, 0xa277e0, 0xc42063a300, 0x0, 0x0)
        /go/src/github.com/Azure/azure-storage-azcopy/cmd/syncDownloadEnumerator.go:165 +0xeeb
github.com/Azure/azure-storage-azcopy/cmd.(*syncDownloadEnumerator).enumerate(0xc420672000, 0xc420650000, 0xc4206674d8, 0x28)
        /go/src/github.com/Azure/azure-storage-azcopy/cmd/syncDownloadEnumerator.go:460 +0x488
github.com/Azure/azure-storage-azcopy/cmd.(*cookedSyncCmdArgs).process(0xc420650000, 0xd7, 0xc420063b48)
        /go/src/github.com/Azure/azure-storage-azcopy/cmd/sync.go:440 +0x373
github.com/Azure/azure-storage-azcopy/cmd.init.13.func2(0xc420144500, 0xc420618210, 0x2, 0x3)
        /go/src/github.com/Azure/azure-storage-azcopy/cmd/sync.go:485 +0x17b
github.com/Azure/azure-storage-azcopy/vendor/github.com/spf13/cobra.(*Command).execute(0xc420144500, 0xc420618180, 0x3, 0x3, 0xc420144500, 0xc420618180)
        /go/src/github.com/Azure/azure-storage-azcopy/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/Azure/azure-storage-azcopy/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xd1ff40, 0x100, 0x56, 0x18)
        /go/src/github.com/Azure/azure-storage-azcopy/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/Azure/azure-storage-azcopy/vendor/github.com/spf13/cobra.(*Command).Execute(0xd1ff40, 0xd, 0xd)
        /go/src/github.com/Azure/azure-storage-azcopy/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/Azure/azure-storage-azcopy/cmd.Execute(0xc420024720, 0xd, 0xc420024720, 0xd)
        /go/src/github.com/Azure/azure-storage-azcopy/cmd/root.go:71 +0x7d
main.main()
        /go/src/github.com/Azure/azure-storage-azcopy/main.go:55 +0x1c8
===error end====

### How can we reproduce the problem in the simplest way?
We have nearly 1855973 blob units. It crashes every time I run this command.

### Have you found a mitigation/solution?
Instead of using I tried to use azcopy cp --overwrite
However it seems like overwrite doesn't work. Instead of consuming total of 2TB of data it consumed 14.47 TB data it caused us to pay 7X more amount to azure :( 
==start of constantcopy.sh==
while :
    do
        ./azcopy cp "https://{Redacted}.blob.core.windows.net/parse/*?{Redacted sas url}" photos/ --recursive=true --overwrite >testson20surekli.txt 2>&1
done
==end of constantcopy.sh==
nohup sh constantcopy.sh &
So there is no practical solution to this problem.

**Comment by zezha-msft** [comment]
Hi @ozgurozkan123, I'm really sorry for the inconvenience.

I've rewrote the sync command and will release it soon.

In the mean time, if you want to avoid overwrite with cp, please do `--overwrite=false`, because `--overwrite` is setting it to true.

**Comment by zezha-msft** [comment]
@ozgurozkan123 sorry for the inconvenience, and I really appreciate your patience!

I've rewritten the sync command and released it in 10.0.8. This issue should be fixed. Please let us know if you still encounter any problem.

Here is a [wiki](https://github.com/Azure/azure-storage-azcopy/wiki/Sync-command-guide) explaining how the new sync command works. Please feel free to reach out if you have any question or concern.

**Comment by DavidLafond** [comment]
What is the difference between using the sync vs copy with overwrite=false?
In my case, I'm unable to make sync work from Blob to Local. I have been able to do a copy from Blob to local of 4 million files, it stopped maybe internet glitch. I tried to use the Sync to continue the job and it scan and mention that everything is sync but I'm missing like 3 million files.  Now I started the copy again with --overwrite=false and look like it will work after 10 minutes 2M files skipped.
